sources/org.osbuild.librepo

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python3
+"""
+Source for downloading rpms using librepo.
+
+Download the list of rpms using a metalink or mirrorlist URL, trying new
+mirrors if there is an error. The files are written to the osbuild file cache
+using the hash as the filename.
+
+It can download files that require secrets. The only secret provider currently
+supported is `org.osbuild.rhsm` for downloading Red Hat content that requires a
+subscriptions.
+"""
+
+import sys
+from typing import Dict
+
+import librepo
+
+from osbuild import sources
+from osbuild.util.rhsm import Subscriptions
+
+# NOTE: The top level schema properties are limited to items and options by the
+# v2 schema definition
+SCHEMA_2 = """
```

Comment:
Some inconsistent indenting in this schema. If we have the same in other modules feel free to ignore.